### PR TITLE
xd75re Multi-layer keymap

### PR DESCRIPTION
**Pull request by tomswartz07**
## Description
Add new xd75re keymap and layout.

This layout has been used and tested on several different xd75re boards and allows for preonic-style layers (raise, lower, adjust)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xd75/keymaps/tomswartz07/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2017 Benjamin Kesselring
```
Comment (fauxpark) [inline keyboards/xd75/keymaps/tomswartz07/config.h:1]: This file can be deleted.

File: keyboards/xd75/keymaps/tomswartz07/keymap.c [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (fauxpark) [inline keyboards/xd75/keymaps/tomswartz07/keymap.c:159]:
```suggestion
```

File: keyboards/xd75/keymaps/tomswartz07/readme.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# The default keymap for xd75, with led controls
```
Comment (fauxpark) [inline keyboards/xd75/keymaps/tomswartz07/readme.md:1]: This file can be deleted, or updated to describe your keymap.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/xd75/keymaps/tomswartz07/keymap.c [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            update_tri_layer(_LOWER, _RAISE, _ADJUST);
+          } else {
+            layer_off(_RAISE);
+            update_tri_layer(_LOWER, _RAISE, _ADJUST);
+          }
+          return false;
+          break;
+  }
+  return true;
+}
```
Comment (Erovia) [inline keyboards/xd75/keymaps/tomswartz07/keymap.c:147]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
}
```

You might be able to achieve the same result with much less code.
Can you check if it works for you.

**Review by Erovia — APPROVED** [review]
LGTM, check if my recommendation works for you.

File: keyboards/xd75/keymaps/tomswartz07/keymap.c (hunk 4 above)
Comment (tomswartz07) [reply keyboards/xd75/keymaps/tomswartz07/keymap.c:147]: Thanks, I'll give it a shot here later today and update as needed.

**Pull request merged**